itou/utils/apis/sentry.py

```diff
@@ -0,0 +1,48 @@
+import logging
+
+import httpx
+import tenacity
+from dateutil.relativedelta import relativedelta
+from django.conf import settings
+from django.utils import timezone
+
+
+logger = logging.getLogger(__name__)
+
+
+class SentryApiClient:
+    def __init__(self):
+        self.client = httpx.Client(
+            base_url=f"{settings.API_SENTRY_BASE_URL}/organizations/{settings.API_SENTRY_ORG_NAME}",
+            headers={
+                "Authorization": f"Bearer {settings.API_SENTRY_STATS_TOKEN}",
+                "Content-Type": "application/json",
+            },
+        )
+
+    @tenacity.retry(wait=tenacity.wait_fixed(2), stop=tenacity.stop_after_attempt(8))
+    def _request(self, start, end):
+        params = {
+            "query": '(event.type:"transaction")',
+            "project": settings.API_SENTRY_PROJECT_ID,
+            "dataset": "metrics",
```

Comment:
J'ai l'impression que ce n'est pas utilisé d'après la [doc](https://docs.sentry.io/api/discover/query-discover-events-in-table-format/) ?

Reply:
Je viens de tester.
Étonnamment, le dictionnaire s'enrichit de nouvelles données quand je retire le paramètre `dataset`. J'ai récupéré ces paramètres sur l'interface de Sentry. Je pense que leur API est principalement faite pour leur front. 

Avec :
```
{
    "data": [
        {
            "apdex()": 0.9400000000000,
            "failure_rate()": 0.020000000000000000
        }
    ],
    "meta": {
        "fields": {
            "apdex()": "number",
            "failure_rate()": "percentage"
        },
        "units": {
            "apdex()": null,
            "failure_rate()": null
        },
        "isMetricsData": true,
        "isMetricsExtractedData": false,
        "tips": {},
        "datasetReason": "unchanged",
        "dataset": "metrics"
    }
}
```

Sans :
```
{
    "data": [
        {
            "tuple(('duration', 300))": [
                [
                    "duration",
                    300
                ]
            ],
            "apdex()": 0.9400000000000000,
            "failure_rate()": 0.02000000000000000,
            "project_threshold_config": [
                "duration",
                300
            ]
        }
    ],
    "meta": {
        "fields": {
            "tuple(('duration', 300))": "string",
            "apdex()": "number",
            "failure_rate()": "percentage",
            "project_threshold_config": "string"
        },
        "units": {
            "tuple(('duration', 300))": null,
            "apdex()": null,
            "failure_rate()": null,
            "project_threshold_config": null
        },
        "isMetricsData": false,
        "isMetricsExtractedData": false,
        "tips": {
            "query": null,
            "columns": null
        },
        "datasetReason": "unchanged",
        "dataset": "discover"
    }
}
```

Reply:
Mais tu as raison, mieux vaut rester sur ce qui est expliqué dans leur doc.